ui/src/app/categories/page.tsx

```diff
@@ -21,58 +21,79 @@ export interface Category {
+    const [categories, setCategories] = useState<Category[]>([]);
+    const [userEmail, setUserEmail] = useState("");
 
-    return (
-        <Box sx={{ padding: '16px 16px 0' }}>
-            <Typography
-                variant="h4"
-                gutterBottom
-                sx={{
-                    color: 'text.primary',
-                    fontWeight: 'bold',
-                }}
-                >
-                Categories
-            </Typography>
+    const fetchCategories = async (email: string) => {
+        try {
+            const response = await fetch(`${process.env.NEXT_PUBLIC_API_ENDPOINT}/category/${email}`);
+            if (!response.ok) {
+                console.error("Failed to fetch categories");
+                return;
+            }
+            const data = await response.json();
+            setCategories(data.data);
+        } catch (error) {
+            console.error("Error fetching categories:", error);
+        }
+    };
 
-            {data &&
-                < TableContainer component={Paper} sx={{ backgroundColor: 'background.paper', borderRadius: '8px'}}>
-                    <Table>
-                        <TableHead
-                            sx={{backgroundColor: 'background.default'}}>
-                            <TableRow>
-                                <StyledTableCell>Name</StyledTableCell>
-                                <StyledTableCell>Email</StyledTableCell>
-                                <StyledTableCell>Currency</StyledTableCell>
-                                <StyledTableCell>Created At</StyledTableCell>
-                            </TableRow>
-                        </TableHead>
-                        <TableBody>
-                            {data.data.map((category: Category) => (
-                                <TableRow
-                                    key={category.id}
-                                    sx={{
-                                        '&:hover': {
-                                            backgroundColor: 'primary.light',
-                                            transition: 'background-color 0.3s ease',
-                                        },
-                                        borderBottom: '1px solid #444',
-                                    }}
-                                >
-                                    <StyledBodyTableCell>{category.name}</StyledBodyTableCell>
-                                    <StyledBodyTableCell>{category.email}</StyledBodyTableCell>
-                                    <StyledBodyTableCell>{category.currency}</StyledBodyTableCell>
-                                    <StyledBodyTableCell>{format(new Date(category.createdAt), 'yyyy-MM-dd')}</StyledBodyTableCell>
-                                </TableRow>
-                            ))}
-                        </TableBody>
-                    </Table>
-                </TableContainer>
+    useEffect(() => {
```

Comment:
Read about retrieving cookies here: https://stackoverflow.com/questions/76595420/how-to-use-cookies-in-client-components-in-next-js-13-and-read-them-on-initial-s, first answer should sufficient to implement. YOu need to create wrapper component of server side.